src/coreclr/src/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -136,15 +137,21 @@ public static int OffsetToStringData
         [MethodImpl(MethodImplOptions.InternalCall)]
         public static extern bool TryEnsureSufficientExecutionStack();
 
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        private static extern object GetUninitializedObjectInternal(
-            // This API doesn't call any constructors, but the type needs to be seen as constructed.
-            // A type is seen as constructed if a constructor is kept.
-            // This obviously won't cover a type with no constructor. Reference types with no
-            // constructor are an academic problem. Valuetypes with no constructors are a problem,
-            // but IL Linker currently treats them as always implicitly boxed.
-            [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors)]
-            Type type);
+        private static unsafe object GetUninitializedObjectInternal(Type type)
+        {
+            RuntimeType rt = (RuntimeType)type;
+            Debug.Assert(rt != null);
+
+            // If somebody asks us to create a Nullable<T>, create a T instead.
+            delegate*<MethodTable*, object> newobjHelper = RuntimeTypeHandle.GetNewobjHelperFnPtr(rt, out MethodTable* pMT, unwrapNullable: true, allowCom: false);
```

Comment:
Would it make sense for the unitialized case to use the cache as well?

Reply:
Possibly, but would need to think on this a bit more. The cache now always attempts to get the fnptrs both for the allocator and for the ctor, failing if an appropriate ctor isn't found. `GetUninitializedObject` can't fail if the ctor isn't found, but I'd like to avoid putting "is the cache half-initialized?" logic back into `Activator`.

Reply:
Fine me. I think it is ok to keep GetUninitializedObject slow as it is today to avoid promoting its use.